logdna/resource_archive.go

```diff
@@ -0,0 +1,278 @@
+package logdna
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const archiveConfigID = "archive"
+
+type archiveConfig struct {
+	Integration        string `json:"integration"`
+	Bucket             string `json:"bucket,omitempty"`
+	Endpoint           string `json:"endpoint,omitempty"`
+	Apikey             string `json:"apikey,omitempty"`
+	Resourceinstanceid string `json:"resourceinstanceid,omitempty"`
+	Accountname        string `json:"accountname,omitempty"`
+	Accountkey         string `json:"accountkey,omitempty"`
+	Projectid          string `json:"projectid,omitempty"`
+	Space              string `json:"space,omitempty"`
+	Accesskey          string `json:"accesskey,omitempty"`
+	Secretkey          string `json:"secretkey,omitempty"`
+	Authurl            string `json:"authurl,omitempty"`
+	Expires            int    `json:"expires,omitempty"`
+	Username           string `json:"username,omitempty"`
+	Password           string `json:"password,omitempty"`
+	Tenantname         string `json:"tenantname,omitempty"`
+}
+
+func resourceArchiveConfigCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	pc := m.(*providerConfig)
+	c := archiveConfig{
```

Comment:
Nit: instead of assigning all of the struct fields indiscriminately, we can modularize this into a function that conditionally assigns the relevant `integration` fields

Reply:
changed!